plugins/out_datadog/datadog.c

```diff
@@ -161,6 +163,22 @@ static int datadog_format(struct flb_config *config,
 
             if (!remapped_tags) {
                 remapped_tags = flb_sds_create_size(byte_cnt);
+                if (!remapped_tags) {
+                    flb_errno();
+                    msgpack_sbuffer_destroy(&mp_sbuf);
+                    msgpack_unpacked_destroy(&result);
+                    return -1;
+                }
+            } else if (flb_sds_len(remapped_tags) < byte_cnt) {
+                tmp = flb_sds_increase(remapped_tags, flb_sds_len(remapped_tags) - byte_cnt);
+                if (!tmp) {
+                    flb_errno();
+                    flb_sds_destroy(remapped_tags);
+                    msgpack_sbuffer_destroy(&mp_sbuf);
+                    msgpack_unpacked_destroy(&result);
+                    return -1;
+                }
+                remapped_tags = tmp;
```

Comment:
So, to confirm:
This new section of code addresses the following problem
1. a remapped_tags buffer is created for the first log event, based on it's size estimations
2. the remapped_tags buffer is cleared and reused for the second log event, disregarding the second logevent's tag size estimation
3. the remapped_tags buffer is too small which triggers a segfault.

The solution is as follows:
Reuse the remapped_tags unless the new size estimate is larger, in which case reinitialize.

Reply:
It seem, though, that this section of code is an optimization and not necessary.

If there is too little space for the contents of the string, then the flb_sds_cat code will reinitialize the buffer with more space for all future use. Reallocating the memory upfront is more efficient, which is why this solution seems like a good idea.

Reply:
> It seem, though, that this section of code is an optimization and not necessary.

> If there is too little space for the contents of the string, then the flb_sds_cat code will reinitialize the buffer with more space for all future use. Reallocating the memory upfront is more efficient, which is why this solution seems like a good idea.

Yea this is what I'm going for... trying to make sure this buffer is reallocated as few times as needed.